sdk/textanalytics/ai-text-analytics/src/textAnalysisClient.ts

```diff
@@ -0,0 +1,490 @@
+   * inferred language is correct.  Scores close to 1 indicate high certainty in
+   * the result.  120 languages are supported.
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/concepts/data-limits}
+   * for data limits.
+   *
+   * ### Examples
+   *
+   * #### Language detection
+   *
+   * ```js
+   * const documents = [<input strings>];
+   * const countryHint = "us";
+   * const results = await client.analyze("LanguageDetection", documents, countryHint);
+   *
+   * for (let i = 0; i < results.length; i++) {
+   *   const result = results[i];
+   *   if (result.error) {
+   *     // a document has an error instead of results
+   *   } else {
+   *     const { name, confidenceScore, iso6391Name } = result.primaryLanguage;
+   *   }
+   * }
+   * ```
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/language-detection/overview}
+   * for more information on language detection.
+   *
+   * @param actionName - the name of the action to be performed on the input
+   *   documents, see ${@link AnalyzeActionName}
+   * @param documents - the input documents to be analyzed
+   * @param options - optional action parameters and settings for the operation
+   *
+   * @returns an array of results where each element contains the primary language
+   *   for the corresponding input document.
+   */
+  public async analyze<ActionName extends "LanguageDetection">(
+    actionName: ActionName,
+    documents: LanguageDetectionInput[],
+    options?: AnalyzeActionParameters<ActionName> & TextAnalysisOperationOptions
+  ): Promise<AnalyzeResult<ActionName>>;
+  /**
+   * Runs a predictive model to determine the language that the passed-in
+   * input strings are written in, and returns, for each one, the detected
+   * language as well as a score indicating the model's confidence that the
+   * inferred language is correct.  Scores close to 1 indicate high certainty in
+   * the result.  120 languages are supported.
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/concepts/data-limits}
+   * for data limits.
+   *
+   * ### Examples
+   *
+   * #### Language detection
+   *
+   * ```js
+   * const documents = [<input strings>];
+   * const countryHint = "us";
+   * const results = await client.analyze("LanguageDetection", documents, countryHint);
+   *
+   * for (let i = 0; i < results.length; i++) {
+   *   const result = results[i];
+   *   if (result.error) {
+   *     // a document has an error instead of results
+   *   } else {
+   *     const { name, confidenceScore, iso6391Name } = result.primaryLanguage;
+   *   }
+   * }
+   * ```
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/language-detection/overview}
+   * for more information on language detection.
+   *
+   * @param actionName - the name of the action to be performed on the input
```

Comment:
should we use analysisName instead of actionName to separate the real-time text ops from the batching/LRO ops?

Reply:
Oh good question, I am not sure tbh. I think one reason to call it `actionName` because the definition of an action is largely the same between the two methods and it could be confusing to call the same thing with different names.

On the other hand, I agree that nothing here says anything about how fast those actions are but I wonder if that really needs to be spelled out.

Reply:
I will mark this as resolved for now since we already have the same discussion in https://github.com/Azure/azure-sdk-for-js/pull/21417#discussion_r853299101